cibuildwheel/windows.py

```diff
@@ -62,20 +70,21 @@ def get_python_configurations(build_selector):
     return python_configurations
 
 
-def extract_zip(zip_src, dest):
+def extract_zip(zip_src: str, dest: str) -> None:
     with ZipFile(zip_src) as zip:
         zip.extractall(dest)
 
 
-def install_cpython(version, arch, nuget):
+def install_cpython(version: str, arch: str, nuget: str) -> str:
     nuget_args = get_nuget_args(version, arch)
     installation_path = os.path.join(nuget_args[-1], nuget_args[0] + '.' + version, 'tools')
     shell([nuget, 'install'] + nuget_args)
     return installation_path
 
 
-def install_pypy(version, arch, url):
+def install_pypy(version: str, arch: str, url: Optional[str]) -> str:
     assert arch == '32'
+    assert url is not None
```

Comment:
Optional and assert? Maybe add to if before call function?

Reply:
I didn't want to break the line `installation_bin_path = install_pypy(python_configuration.version, python_configuration.url)`. But yeah, maybe.

Reply:
I think about ` elif python_configuration.identifier.startswith('pp') and python_configuration.url is not None:

Reply:
Yeah, just did that. Just give me a second to commit :-)

Reply:
Oh, wait no. I did it slightly differently.

But if `python_configuration.identifier.startswith('pp')` and `python_configuration.url is None`, that's a mistake. So we should assert on that.

Reply:
so tests will fail on unsuported python interpreter
Or I have maybe introduce CPythonConfiguration and PyPythonConfiguration? To force at the level of code url?

Reply:
What? No. `python_configuration.url` is something we list ourselves. How can this fail?

Reply:
This is literally what asserts are made for, right? To check programmatic assumptions that are not user input.

Reply:
(Or I'm really missing something, and then I'm curious what obvious thing I overlooked. Sorry, then, for being too unbelieving :-) )